src/models/BaseModel.py

```diff
@@ -1,118 +1,205 @@
-        BaseModel.normalized_location_data = pd.read_csv(train_normalized_location_file_name)
+    def export_data(list_to_write, file_name):
+        print('Exporting data to ../data/'+file_name)
+        print('Will read from there next time to avoid pre-processing')
+        f = open('../data/'+file_name+'.csv', 'a')
+        for main_index, sub_list in enumerate(list_to_write):
+            for index, item in enumerate(sub_list):
+                f.write(str(item))
+                if index < len(sub_list)-1:
+                    f.write(',')
+            if main_index < len(list_to_write)-1:
+                f.write('\n')
+        f.close()
+        print('Exporting complete \n')
 
     @staticmethod
-    def load_csv_data(train_data_csv_file_name):
-        BaseModel.data = pd.read_csv(train_data_csv_file_name)
+    def export_prediction(prediction_to_write, file_name):
+        print('Exporting prediction to ../predictions/'+file_name)
+        f = open('../predictions/'+file_name+'.csv', 'a')
+        for index, item in enumerate(prediction_to_write):
+            f.write(str(item))
+            if index < len(prediction_to_write)-1:
+                f.write('\n')
+        f.close()
+        print('Exporting complete \n')
 
     @staticmethod
-    def load_all_data(self):
+    def load_all_data():
         """
         loads all data into the static set
-        useful, if this ever is going to be parallelized
-        (i.e. load data first, then parallelize models)
         """
-        BaseModel.load_csv_data(TRAIN_DATA_CSV_FILE_NAME)
-        BaseModel.load_location_data(TRAIN_NORMALIZED_LOCATION_FILE_NAME)
+        BaseModel.data = pd.read_csv(BaseModel.TRAIN_DATA_CSV_FILE_NAME)
+        BaseModel.normalized_location_data = pd.read_csv(
+            BaseModel.TRAIN_NORMALIZED_LOCATION_FILE_NAME)
 
-    @abc.abstractmethod
-    def predict_salary(self):
-        """
-        abstract method, where actual prediction will be implemented
+    @staticmethod
+    def load_cleaned_data():
+        loaded_data = pd.read_csv(BaseModel.CLEANED_FILE_NAME, header=None)
+        loaded_data = loaded_data.as_matrix()
+        loaded_data = loaded_data.astype(int)
+        return loaded_data
+
+    abc.abstractmethod
 
-        :return: list of predictions for test set
+    def predict(self):
+        """
+        #abstract method, where actual prediction will be implemented
+        #:return: list of predictions for test set
         """
         raise NotImplementedError
 
     def run(self):
         """
         predict salary according to implementation of model and calculate the error
-
         :return error of model
         """
-        salary = self.predict_salary()
-        error = self.calculate_error(salary)
-        return error
+        print('Training begins')
+        (train_result, test_result) = self.predict()
+        (train_error, test_error) = self.calculate_error(train_result, test_result)
+        return 1
```

Comment:
Will prob return test_results to use in esembles in the future but if i added the infrastructure to do ensembles it would be too much for a 1 PR, it was 1, left it as 1 for now